src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -51,32 +53,73 @@ public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtract
         return pdpFileSuccessfullyCreated;
     }
 
-    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile) {
+    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile, ConcurRequestExtractBatchReportData reportData) {
         int totalPdpDetailRecordsCount = 0;
         KualiDecimal totalPdpDetailRecordsAmount = KualiDecimal.ZERO;
+        int totalClonedCashAdvanceLinesCount = 0;
+        KualiDecimal totalClonedCashAdvanceLinesAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesUseInExpenseReportCount = 0;
+        KualiDecimal totalCashAdvancesUseInExpenseReportAmount= KualiDecimal.ZERO;
+        int totalDuplicateCashAdvanceCount = 0;
+        KualiDecimal totalDuplicateCashAdvanceAmount = KualiDecimal.ZERO;
+        int totalNonCashAdvanceRecordsCount = 0;
+        KualiDecimal totalNonCashAdvanceRecordsAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesNotProcessedDueToValidationErrorsCount = 0;
+        KualiDecimal totalCashAdvancesNotProcessedDueToValidationErrorsAmount = KualiDecimal.ZERO;
+        int totalProcessingAttemptedLineCount = 0;
+        KualiDecimal totalProcessingAttemptedLineAmount = KualiDecimal.ZERO;
```

Comment:
I'm curious,  It looks like you are setting these values in this function, and then just update the corresponding values in the report data object at the end.  What does the local copy of these values buy us?  Would it make more sense just to use the values in the report data object?